src/cpp/session/modules/SessionDiagnostics.cpp

```diff
@@ -866,12 +867,43 @@ Error lintRSourceDocument(const json::JsonRpcRequest& request,
             documentId,
             isExplicit,
             isFragment);
-   
-   pResponse->setResult(lintAsJson(results.lint()));
-   
+
+   std::vector<module_context::SourceMarker> markers =
+       modules::markers::markersForFile(documentPath);
+   LintItems lintItems = results.lint();
+
+   using namespace module_context;
+   for (SourceMarker marker : markers)
+   {
+      LintType markerLintType;
+      switch (marker.type)
+      {
+      case SourceMarker::Type::Error:
+         markerLintType = LintTypeError;
+         break;
+      case SourceMarker::Type::Warning:
+         markerLintType = LintTypeWarning;
+         break;
+      case SourceMarker::Type::Info:
+         markerLintType = LintTypeInfo;
+         break;
+      case SourceMarker::Type::Style:
+         markerLintType = LintTypeStyle;
+         break;
+      default:
+         markerLintType = LintTypeInfo;
+      }
+
+      int line = marker.line - 1; // markers begin the index at 1 and lint items begin at 0
```

Comment:
Should we add any checks here? What would happen if `marker.line == 0` for some reason?

Reply:
I'll add one just in case. An invalid line number links to line 1 and the source marker pane doesn't show a line number if it is negative.